src/main/java/rx/plugins/RxJavaHooks.java

```diff
@@ -0,0 +1,944 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.plugins;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+
+import rx.*;
+import rx.Completable.CompletableOnSubscribe;
+import rx.Observable.*;
+import rx.annotations.Experimental;
+import rx.functions.*;
+import rx.internal.operators.*;
+
+/**
+ * Utility class that holds hooks for various Observable, Single and Completable lifecycle-related
+ * points as well as Scheduler hooks.
+ * <p>
+ * The class features a lockdown state, see {@link #lockdown()} and {@link #isLockdown()}, to
+ * prevent further changes to the hooks. 
+ */
+@Experimental
+public final class RxJavaHooks {
+    /** Utility class. */
+    private RxJavaHooks() {
+        throw new IllegalStateException("No instances!");
+    }
+    
+    /**
+     * Prevents changing the hook callbacks when set to true.
+     */
+    /* test */ static volatile boolean lockdown;
```

Comment:
To test if the lockdown actually works, this has to be locked down but then unlocked to allow other tests setting their hooks after that.